apps/web/src/app/(dashboard)/documents/data-table.tsx

```diff
@@ -62,7 +63,9 @@ export const DocumentsDataTable = ({
           {
             header: 'Created',
             accessorKey: 'createdAt',
-            cell: ({ row }) => <LocaleDate date={row.original.createdAt} />,
+            cell: ({ row }) => (
+              <LocaleDate date={row.original.createdAt} format={DateTime.DATETIME_SHORT} />
```

Comment:
Tested it locally and LGTM

- Date comes in 12hr format 
- Is in line with our Timestamp design according to our design file

<img width="517" alt="Screenshot 2024-05-08 at 3 23 38 PM" src="https://github.com/documenso/documenso/assets/23498248/006b3120-e2d8-4065-afab-5a3cb9a51350">